source/features/add-co-authored-by.js

```diff
@@ -0,0 +1,125 @@
+import select from 'select-dom';
+import * as api from '../libs/api';
+import {escapeForGql} from '../libs/utils';
+import {getOwnerAndRepo, getPRNumber} from '../libs/page-detect';
+
+const coAuthorData = {};
+
+async function fetchCoAuthoredData() {
+	const {ownerName, repoName} = getOwnerAndRepo();
+	const prNumber = getPRNumber();
+
+	if (!ownerName || !repoName || !prNumber) {
+		return;
+	}
+
+	const {data: contributorData} = await api.v4(
+		`{
+			repository(owner: "${ownerName}", name: "${repoName}") {
+				pullRequest(number: ${prNumber}) {
+					commits(first: 100) {
+						nodes {
+							commit {
```

Comment:
`commit` actually has an `email` field directly from the commit, so we don't need to generate it. Example of a user that has a private email but his email appears in a commit:

<img width="237" alt="" src="https://user-images.githubusercontent.com/1402241/50326189-6c418580-051b-11e9-9024-41d9d1726521.png">

Reply:
It's possible to commit without an email address (eg, `git commit --author "Anon <>"`), so I'm not sure if this field can be relied upon.

Reply:
We should still probably prefer it over the one in the profile when present.

Reply:
Yes, we should always prefer the commit email as that's what the user has assigned to use for commits and credit.

Reply:
👍🏻